joblib/hashing.py

```diff
@@ -189,7 +189,10 @@ def save(self, obj):
         if isinstance(obj, self.np.ndarray) and not obj.dtype.hasobject:
             # Compute a hash of the object
             # The update function of the hash requires a c_contiguous buffer.
-            if obj.flags.c_contiguous:
+            # Od arrays have to be flattened.
```

Comment:
I would put the comment inside the if clause and I would explain why, e.g. "0d arrays need to be flattened because viewing them as bytes raises an exception".